packages/app/cypress/e2e/runner/runner.ui.cy.ts

```diff
@@ -94,6 +94,33 @@ describe('src/cypress/runner', () => {
       .should('have.class', 'command-is-pinned')
     })
 
+    it('correctly resets highlight toggle state when pinning new command', () => {
```

Comment:
You fixed the `Before | After` sync as well, can you write a test for that?

Reply:
Tyler did here https://github.com/cypress-io/cypress/pull/20670/commits/074a97432504b8c3af0536c86c898e45f509c11a#diff-14b335c5128675493e52a73e003be1d2ae2013934caa565075a40ef4361fbcffR129